utils/shmem.c

```diff
@@ -0,0 +1,92 @@
+#include <errno.h>
+#include <fcntl.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "shmem.h"
```

Comment:
Nit: I keep the include path from the root directory even if it's in the same directory.  I think it's better for maintenance.  So please change it to `"utils/shmem.h"`.